### PR TITLE
Nixify

### PR DESCRIPTION
**Pull request by x10an14-nav**
Originally set to draft due to intent of proving below claims with a long gist of some sort:
1. Docker image size reduced by half (from ~17mb to ~7)
1. Build time (of docker image) reduced by almost half (~40+seconds)
1. Dev env now reproducible
1. Docker image can now be built not only repeatably, but reproducibly

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+  inputs = {
+    crane = {
+      url = "github:ipetkov/crane";
+      inputs = {
+        nixpkgs.follows = "nixpkgs";
+        flake-utils.follows = "flake-utils";
+      };
+    };
+    flake-utils.url = "github:numtide/flake-utils";
+    nixpkgs.url = "github:NixOS/nixpkgs";
```
Comment (Reasonable-Solutions) [inline flake.nix:13]: I would pin this to some revision.

File: README.md [hunk 2]
```diff
@@ -10,15 +10,20 @@ Things to mutate
...
-Mutilator uses [earthly](https://earthly.dev) for building. 
+Mutilator can use [earthly](https://earthly.dev) for building.
 If you don't have earthly installed, you can use the wrapper at `./earthlyw`, which downloads the latest version for you.
 
 * `earthly ls` to list targets
 * `earthly +docker` to build primary target
 * `earthly +aiven-types` to generate rust models for Aiven CRDs (see below)
 
+## Building Docker Image w/Nix
+1. Use `earthly +aiven-types` to generate rust models
```
Comment (Reasonable-Solutions) [inline README.md:23]: the pragmatic approach?

File: flake.nix [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        targets = [
+          buildTarget
+          (pkgs.rust.toRustTargetSpec pkgs.stdenv.hostPlatform)
+        ];
+      };
+
+      # Set-up build dependencies and configure rust
+      craneLib = (crane.mkLib pkgs).overrideToolchain rustToolchain;
+
+      # Shamelessly stolen from:
```
Comment (Reasonable-Solutions) [inline flake.nix:50]: To an explain on why src filtering is a good idea.

File: .envrc [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      export "$line"
+    done < "$file"
+  done
+}
+
+# Export all .gitignored (potentially) secret environment variables
+files="$(find . -maxdepth 1 -type f -name '.env*' -and -not -name '.envrc')"
+load_envs "$files"
+
+# Load nix env for all the cool people
```
Comment (Reasonable-Solutions) [inline .envrc:32]: 😎

**Comment by x10an14-nav** [comment]
Now, with the last two commits, the Nixify branch also improves the code initiating the logging system.

**Review by Reasonable-Solutions — APPROVED** [review]

**Pull request merged**